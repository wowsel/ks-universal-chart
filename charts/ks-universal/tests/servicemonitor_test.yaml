suite: test servicemonitor
templates:
  - deployment.yaml
tests:
  - it: should create a service monitor when autoCreateServiceMonitor is enabled
    set:
      deployments:
        test-app:
          containers:
            main:
              image: nginx
              imageTag: latest
              ports:
                http:
                  containerPort: 80
                http-metrics:
                  containerPort: 8080
          autoCreateService: true
          autoCreateServiceMonitor: true
          serviceMonitor:
            interval: "15s"
            path: "/metrics"
    asserts:
      - matchSnapshot: {} 