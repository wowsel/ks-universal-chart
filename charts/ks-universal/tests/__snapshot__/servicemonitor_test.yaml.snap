should create a service monitor when autoCreateServiceMonitor is enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-0.2.14
      name: test-app
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
      template:
        metadata:
          labels:
            app.kubernetes.io/component: test-app
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            helm.sh/chart: ks-universal-0.2.14
        spec:
          containers:
            - image: nginx:latest
              name: main
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
                - containerPort: 8080
                  name: http-metrics
                  protocol: TCP
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-0.2.14
      name: test-app
    spec:
      endpoints:
        - interval: 15s
          path: /metrics
          port: http-metrics
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
